src/settings/mod.rs

```diff
@@ -31,33 +30,34 @@ use crate::warn_user_once;
 
 pub mod configuration;
 pub mod flags;
+pub mod hashable;
 pub mod options;
 pub mod options_base;
 pub mod pyproject;
 pub mod types;
 
 const CARGO_PKG_VERSION: &str = env!("CARGO_PKG_VERSION");
 
-#[derive(Debug)]
+#[derive(Debug, Hash)]
 #[allow(clippy::struct_excessive_bools)]
 pub struct Settings {
-    pub allowed_confusables: FxHashSet<char>,
+    pub allowed_confusables: BTreeSet<char>,
     pub builtins: Vec<String>,
     pub cache_dir: PathBuf,
-    pub dummy_variable_rgx: Regex,
-    pub enabled: FxHashSet<RuleCode>,
-    pub exclude: GlobSet,
-    pub extend_exclude: GlobSet,
-    pub external: FxHashSet<String>,
+    pub dummy_variable_rgx: HashableRegex,
+    pub enabled: BTreeSet<RuleCode>,
```

Comment:
These are all intentionally using `FxHashSet`. Last I checked it actually made a measurable difference in the benchmarks.

Reply:
To be clear: I don't want to change this right now if it has any noticeable impact on perf. We read from these _extremely_ often (but only hash the configuration once).

However, in theory, this could be even better, if that's interesting to you: https://github.com/charliermarsh/ruff/issues/703

Reply:
Right, I agree that we shouldn't change this if it makes a measurable difference ... I'll do a benchmark.

If it makes a difference I'll just amend this PR to introduce `HashableHashSet` and `HashableHashMap` wrapper types instead.

Reply:
Ok yes it does make a difference ... good catch! I'll change it to use newtypes instead after #1891 has been merged.

Reply:
I'm actually shocked at how big of an impact it has.

Before:

![Screen Shot 2023-01-15 at 3 20 20 PM](https://user-images.githubusercontent.com/1309177/212564931-b0c2ac73-b0e0-4c45-b659-8e3695ac0fba.png)

After:

![Screen Shot 2023-01-15 at 3 20 08 PM](https://user-images.githubusercontent.com/1309177/212564933-b2ee329b-1a0b-4d66-a299-a7b01c156be2.png)

Reply:
Maybe it shouldn't be a surprise given that we're doing these `enabled` lookups _constantly_.

Reply:
I'm going to revisit using a bitset for this... some early benchmarks suggest it could provide another 10% speedup.

Reply:
I am actually doing that just now ... PR incoming.

Reply:
Awesome.